modules/otel-collector-components.adoc

```diff
@@ -199,6 +199,152 @@ The OpenCensus receiver provides backwards compatibility with the OpenCensus pro
 To match any origin, enter only `+*+`.
 
+[id="k8scluster-receiver_{context}"]
+=== Kubernetes cluster receiver
+
+The Kubernetes cluster receiver is responsible for gathering cluster-level metrics and entity events from the Kubernetes API server. It utilizes the K8s API to stay informed about updates. Authentication for this receiver is exclusively supported through service accounts.
+
+.OpenTelemetry Collector custom resource with the enabled Kubernetes cluster receiver
+[source,yaml]
+----
+  receivers:
+    k8s_cluster:
+      collection_interval: 10s
+  exporters:
+    debug:
+  service:
+    pipelines:
+      metrics:
+        receivers: [k8s_cluster]
+        exporters: [debug]
+      logs/entity_events:
+        receivers: [k8s_cluster]
+        exporters: [debug]
+----
+
+This receiver also requires to configure a Service Account with some RBAC rules
+
+.Kubernetes cluster receiver service account
+[source,yaml]
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    app: otelcontribcol
+  name: otelcontribcol
+----
+
+.Kubernetes cluster receiver service account RBAC
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: otelcontribcol
+  labels:
+    app: otelcontribcol
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - events
+  - namespaces
+  - namespaces/status
+  - nodes
+  - nodes/spec
+  - pods
+  - pods/status
+  - replicationcontrollers
+  - replicationcontrollers/status
+  - resourcequotas
+  - services
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - apps
+  resources:
+  - daemonsets
+  - deployments
+  - replicasets
+  - statefulsets
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - extensions
+  resources:
+  - daemonsets
+  - deployments
+  - replicasets
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - batch
+  resources:
+  - jobs
+  - cronjobs
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+    - autoscaling
+  resources:
+    - horizontalpodautoscalers
+  verbs:
+    - get
+    - list
+    - watch
+----
+
+And the ClusterRoleBinding to bind the RBAC with the ServiceAccount
+.Kubernetes ClusterRoleBinding resource
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: otelcontribcol
+  labels:
+    app: otelcontribcol
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: otelcontribcol
+subjects:
+- kind: ServiceAccount
+  name: otelcontribcol
+  namespace: default
+----
+
+You can enable OpenShift support to collect OpenShift specific metrics in addition to the default Kubernetes ones. To do this,set the distribution key to openshift.
+
+.OpenTelemetry Collector custom resource with the enabled Kubernetes cluster receiver with openshift distribution
+[source,yaml]
+----
+  k8s_cluster:
+    distribution: openshift
+----
+
+Also if you enable OpenShift distribution you need to add the following rules to your ClusterRole.
```

Comment:
@rubenvp8510, can I incorporate these lines into this ClusterRole: https://github.com/openshift/openshift-docs/pull/75604/files#diff-294ec585891d4f53f51203c945436edac35259850eb19f99f502661759203d21R242? Or does this require another ClusterRole?

Reply:
:slightly_smiling_face: I mean the downstream OTEL product is only supported on OpenShift, so we have to make the OpenShift configuration the default settings rather than optional.

Reply:
It will be the same cluster role.

Reply:
Thank you, Ruben. I'm incorporating it then.